src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -786,38 +933,31 @@ evalApp γ ctx e0 es et
   , Just eq <- Map.lookup f (knAms γ)
   , length (eqArgs eq) <= length es
   = do
-       env  <- gets (seSort . evEnv)
+       env <- gets (seSort . evEnv) 
        okFuel <- checkFuel f
-       if okFuel && et /= FuncNormal
-         then do
-                let (es1,es2) = splitAt (length (eqArgs eq)) es
-                    newE = substEq env eq es1
-                (e', fe) <- evalIte γ ctx et newE -- TODO:FUEL this is where an "unfolding" happens, CHECK/BUMP counter
-                let e2' = stripPLEUnfold e'
-                    e3' = simplify γ ctx (eApps e2' es2) -- reduces a bit the equations
-                    undecidedGuards = case e' of
-                      EIte{} -> True
-                      _ -> False
-
-                if undecidedGuards
-                  then do
-                    modify $ \st ->
-                      st {
-                        evPendingUnfoldings = M.insert (eApps e0 es) e3' (evPendingUnfoldings st)
-                      }
-                    -- Don't unfold the expression if there is an if-then-else
-                    -- guarding it, just to preserve the size of further
-                    -- rewrites.
-                    return (Nothing, noExpand)
-                  else do
-                    useFuel f
-                    modify $ \st ->
-                      st
-                        { evNewEqualities = S.insert (eApps e0 es, e3') (evNewEqualities st)
-                        , evPendingUnfoldings = M.delete (eApps e0 es) (evPendingUnfoldings st)
-                        }
-                    return (Just $ eApps e2' es2, fe)
-         else return (Nothing, noExpand)
+       if okFuel && et /= FuncNormal then do
+        let (es1, es2) = splitAt (length (eqArgs eq)) es
+        -- Doing elaboration ahead of time would cause too much
+        -- monomorphization
```

Comment:
I'm guessing every reader of this comment will want to know:
1. for what purpose monomorphization would be too much,
2. what does monomorphization mean in this context
3. what is the interaction between elaboration and whatever purpose is claimed in (1).

Reply:
That answers (2) well, and (1) has something to do with eta-expansion. 

> ... and during eta expansion will cause typing problems

What is the typing problem that you are referring to?

Reply:
When you perform the replacement the equation before being sent to the SMT solver will be elaborated and it will fail since the unification will fail

Reply:
I don't think I follow your explanation. Could you make an example to illustrate perhaps?

Reply:
If you have an equation for id if you run unify on it's body (meaning on the field eqBody of the Equation) you will get all the variables annotates with the monomorphized type `a#foo` and not with the liquid type variable `@foo` so when you perform the substitution in some place where the type variable is instantiated to some concrete type 
Bar, the evalCandsLoop will typecheck and complain that it cannot unify `a#foo` with `Bar` instead if we do it later and we can keep the liquid type variable during typechecking `@foo` is unified with `Bar`

Reply:
In addition, the comment should probably explain why elaborating the term is necessary at this point.